{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Manipulation of Trans Proteomic Pipeline (TPP) Comet database search results of *Prochlorococus MED4* LC-MS/MS data using python.\n",
    "\n",
    "Starting with: \n",
    "\n",
    "- Comet output (.xlxs and .csv) of PTM-optimized database searches, sorted by XCorr (descending) and run through XInteract to extract precursor intensities and protein descriptions mapped from the search database.\n",
    "\n",
    "Ending with:\n",
    "\n",
    "- Files with stripped (no PTMs or tryptic ends) peptide lists and\n",
    "- Columns with #'s of each modification in every sequence\n",
    "- Column with stripped peptide lengths (# amino acids)\n",
    "- Histogram of sequence lengths\n",
    "- Bar plots of PTM occurance\n",
    "\n",
    "### To use:\n",
    "\n",
    "#### 1. Change the input file name in *IN 4*\n",
    "#### 2. Change output file name in *IN 6*, *IN 7*, *IN 8*"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For technical duplicates, I exported Comet search results as both Excel files and as CSVs into my ETNP 2017 git repo:\n",
    "\n",
    "Also, when running through XInteract in the TPP, I combined the duplicate injections into a single PepXML file which I exported as an xls file and converted to a csv."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/RAL4_95_MED2_trypsin/TPP\n"
     ]
    }
   ],
   "source": [
    "cd /home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/RAL4_95_MED2_trypsin/TPP/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " RAL4_MED2_combine_Comet2.5Xcorr_proteins.txt\r\n",
      " RAL4_MED2_combine_Comet3Xcorr_proteins.txt\r\n",
      " RAL4_MED2_trypsin_1_PTMopt_Comet.csv\r\n",
      " RAL4_MED2_trypsin_1_PTMopt_Comet_stripped.csv\r\n",
      " RAL4_MED2_trypsin_1_PTMopt_Comet_stripped_peptides_2.5XCorr.txt\r\n",
      " RAL4_MED2_trypsin_1_PTMopt_Comet_stripped_peptides.txt\r\n",
      " RAL4_MED2_trypsin_1_PTMopt_Comet_stripped_work.ods\r\n",
      " RAL4_MED2_trypsin_1_PTMopt_Comet_unfiltered.csv\r\n",
      " RAL4_MED2_trypsin_1_PTMopt_Comet.xlsx\r\n",
      " RAL4_MED2_trypsin_1_PTMopt_PepProp90.csv\r\n",
      " RAL4_MED2_trypsin_1_PTMopt_PepProp90_stripped.csv\r\n",
      " RAL4_MED2_trypsin_1_PTMopt_PepProp90_stripped_peptides\r\n",
      " RAL4_MED2_trypsin_1_PTMopt_PepProp90.xlsx\r\n",
      " RAL4_MED2_trypsin_2_PTMopt_Comet.csv\r\n",
      " RAL4_MED2_trypsin_2_PTMopt_Comet_stripped.csv\r\n",
      " RAL4_MED2_trypsin_2_PTMopt_Comet_stripped_peptides_2.5XCorr.txt\r\n",
      " RAL4_MED2_trypsin_2_PTMopt_Comet_stripped_peptides.txt\r\n",
      " RAL4_MED2_trypsin_2_PTMopt_Comet_stripped_work.ods\r\n",
      " RAL4_MED2_trypsin_2_PTMopt_Comet.xlsx\r\n",
      " RAL4_MED2_trypsin_2_PTMopt_PepProp90.csv\r\n",
      " RAL4_MED2_trypsin_2_PTMopt_PepProp90_stripped.csv\r\n",
      " RAL4_MED2_trypsin_2_PTMopt_PepProp90_stripped_peptides\r\n",
      " RAL4_MED2_trypsin_2_PTMopt_PepProp90.xlsx\r\n",
      " RAL4_MED2_trypsin_combine_PTMopt_Comet_3XCorr_noDECOY.csv\r\n",
      " RAL4_MED2_trypsin_combine_PTMopt_Comet_3XCorr_noDECOY.txt\r\n",
      " RAL4_MED2_trypsin_combine_PTMopt_Comet_stripped__nonredundant_peptides_2.5XCorr.txt\r\n",
      " RAL4_MED2_trypsin_combine_PTMopt_Comet_stripped_peptides_2.5XCorr.txt\r\n",
      " RAL4_MED2_trypsin_combine_PTMopt_Comet_stripped_peptides_3XCorr.txt\r\n",
      " RAL4_MED2_trypsin_combine_PTMopt_Comet_stripped_peptides.txt\r\n",
      " RAL95_MED2_trypsin_combine_Comet25_totals.csv\r\n",
      " RAL95_MED2_trypsin_combine_Comet3_totals.csv\r\n",
      " RAL95_MED2_trypsin_combine_Comet_totals.csv\r\n",
      " RAL95_MED2_trypsin_combine_PTMopt_DN50_nonredundant_stripped_peptides.txt\r\n",
      " RAL95_MED2_trypsin_combine_PTMopt_DN80_nonredundant_stripped_peptides.txt\r\n",
      " ral_95_med2_trypsin_combine_quant_concat_pepxml.csv\r\n",
      "'ral_95_med2_trypsin_combine_quant.pep 2b.xls'\r\n",
      " ral_95_med2_trypsin_combine_quant.pep.sort.csv\r\n",
      " ral_95_med2_trypsin_combine_quant.pep.srt.xls\r\n",
      " ral_95_med2_trypsin_combine_quant.pep.xls\r\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'1.0.5'"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# LIBRARIES\n",
    "#import pandas library for working with tabular data\n",
    "import os\n",
    "os.getcwd()\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "#from matplotlib import pyplot\n",
    "import seaborn as sns\n",
    "from scipy.stats import kde\n",
    "#import regular expresson (regex)\n",
    "import re\n",
    "#check pandas version\n",
    "pd.__version__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# redundant Comet peptides in combined dataframe 65535\n",
      "# nonredundant Comet peptides in combined dataframe 65535\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>xcorr</th>\n",
       "      <th>deltacn</th>\n",
       "      <th>expect</th>\n",
       "      <th>ions</th>\n",
       "      <th>peptide</th>\n",
       "      <th>protein</th>\n",
       "      <th>calc_neutral_pep_mass</th>\n",
       "      <th>precursor_intensity</th>\n",
       "      <th>protein_descr</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>spectrum</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>022016_RAL4_95_MED2_trypsin_2.32518.32518.4</th>\n",
       "      <td>8.780</td>\n",
       "      <td>1.000</td>\n",
       "      <td>5.070000e-12</td>\n",
       "      <td>12/204</td>\n",
       "      <td>K.LAIDDSSINLDQVDYIN[115.03]AHGTSTTANDKNETSAIK.S</td>\n",
       "      <td>PMM1609</td>\n",
       "      <td>3734.7759</td>\n",
       "      <td>5250000.0</td>\n",
       "      <td>| fabF | 3-oxoacyl-[acyl-carrier-protein] synt...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>022016_RAL4_95_MED2_trypsin_1.41211.41211.4</th>\n",
       "      <td>8.768</td>\n",
       "      <td>1.000</td>\n",
       "      <td>1.420000e-09</td>\n",
       "      <td>22/156</td>\n",
       "      <td>K.LFADENHLSPAVTAIQIEDIDAEQFRK.N</td>\n",
       "      <td>PMM0035</td>\n",
       "      <td>3069.5407</td>\n",
       "      <td>2970000.0</td>\n",
       "      <td>| DHSS | soluble hydrogenase small subunit</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>022016_RAL4_95_MED2_trypsin_2.50643.50643.4</th>\n",
       "      <td>8.599</td>\n",
       "      <td>0.583</td>\n",
       "      <td>2.140000e-12</td>\n",
       "      <td>22/156</td>\n",
       "      <td>R.SGLQNAASIAGM[147.04]VLTTEC[160.03]IVADLPEKK.D</td>\n",
       "      <td>PMM1436</td>\n",
       "      <td>2831.4409</td>\n",
       "      <td>7140000.0</td>\n",
       "      <td>| groEL | chaperonin GroEL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>022016_RAL4_95_MED2_trypsin_1.32793.32793.4</th>\n",
       "      <td>8.578</td>\n",
       "      <td>1.000</td>\n",
       "      <td>9.310000e-10</td>\n",
       "      <td>18/204</td>\n",
       "      <td>K.LAIDDSSIN[115.03]LDQVDYIN[115.03]AHGTSTTANDK...</td>\n",
       "      <td>PMM1609</td>\n",
       "      <td>3735.7599</td>\n",
       "      <td>6550000.0</td>\n",
       "      <td>| fabF | 3-oxoacyl-[acyl-carrier-protein] synt...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>022016_RAL4_95_MED2_trypsin_1.50751.50751.4</th>\n",
       "      <td>8.469</td>\n",
       "      <td>0.646</td>\n",
       "      <td>3.430000e-14</td>\n",
       "      <td>24/156</td>\n",
       "      <td>R.SGLQNAASIAGM[147.04]VLTTEC[160.03]IVADLPEKK.D</td>\n",
       "      <td>PMM1436</td>\n",
       "      <td>2831.4409</td>\n",
       "      <td>8290000.0</td>\n",
       "      <td>| groEL | chaperonin GroEL</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             xcorr  deltacn        expect  \\\n",
       "spectrum                                                                    \n",
       "022016_RAL4_95_MED2_trypsin_2.32518.32518.4  8.780    1.000  5.070000e-12   \n",
       "022016_RAL4_95_MED2_trypsin_1.41211.41211.4  8.768    1.000  1.420000e-09   \n",
       "022016_RAL4_95_MED2_trypsin_2.50643.50643.4  8.599    0.583  2.140000e-12   \n",
       "022016_RAL4_95_MED2_trypsin_1.32793.32793.4  8.578    1.000  9.310000e-10   \n",
       "022016_RAL4_95_MED2_trypsin_1.50751.50751.4  8.469    0.646  3.430000e-14   \n",
       "\n",
       "                                               ions  \\\n",
       "spectrum                                              \n",
       "022016_RAL4_95_MED2_trypsin_2.32518.32518.4  12/204   \n",
       "022016_RAL4_95_MED2_trypsin_1.41211.41211.4  22/156   \n",
       "022016_RAL4_95_MED2_trypsin_2.50643.50643.4  22/156   \n",
       "022016_RAL4_95_MED2_trypsin_1.32793.32793.4  18/204   \n",
       "022016_RAL4_95_MED2_trypsin_1.50751.50751.4  24/156   \n",
       "\n",
       "                                                                                       peptide  \\\n",
       "spectrum                                                                                         \n",
       "022016_RAL4_95_MED2_trypsin_2.32518.32518.4    K.LAIDDSSINLDQVDYIN[115.03]AHGTSTTANDKNETSAIK.S   \n",
       "022016_RAL4_95_MED2_trypsin_1.41211.41211.4                    K.LFADENHLSPAVTAIQIEDIDAEQFRK.N   \n",
       "022016_RAL4_95_MED2_trypsin_2.50643.50643.4    R.SGLQNAASIAGM[147.04]VLTTEC[160.03]IVADLPEKK.D   \n",
       "022016_RAL4_95_MED2_trypsin_1.32793.32793.4  K.LAIDDSSIN[115.03]LDQVDYIN[115.03]AHGTSTTANDK...   \n",
       "022016_RAL4_95_MED2_trypsin_1.50751.50751.4    R.SGLQNAASIAGM[147.04]VLTTEC[160.03]IVADLPEKK.D   \n",
       "\n",
       "                                             protein  calc_neutral_pep_mass  \\\n",
       "spectrum                                                                      \n",
       "022016_RAL4_95_MED2_trypsin_2.32518.32518.4  PMM1609              3734.7759   \n",
       "022016_RAL4_95_MED2_trypsin_1.41211.41211.4  PMM0035              3069.5407   \n",
       "022016_RAL4_95_MED2_trypsin_2.50643.50643.4  PMM1436              2831.4409   \n",
       "022016_RAL4_95_MED2_trypsin_1.32793.32793.4  PMM1609              3735.7599   \n",
       "022016_RAL4_95_MED2_trypsin_1.50751.50751.4  PMM1436              2831.4409   \n",
       "\n",
       "                                             precursor_intensity  \\\n",
       "spectrum                                                           \n",
       "022016_RAL4_95_MED2_trypsin_2.32518.32518.4            5250000.0   \n",
       "022016_RAL4_95_MED2_trypsin_1.41211.41211.4            2970000.0   \n",
       "022016_RAL4_95_MED2_trypsin_2.50643.50643.4            7140000.0   \n",
       "022016_RAL4_95_MED2_trypsin_1.32793.32793.4            6550000.0   \n",
       "022016_RAL4_95_MED2_trypsin_1.50751.50751.4            8290000.0   \n",
       "\n",
       "                                                                                 protein_descr  \n",
       "spectrum                                                                                        \n",
       "022016_RAL4_95_MED2_trypsin_2.32518.32518.4  | fabF | 3-oxoacyl-[acyl-carrier-protein] synt...  \n",
       "022016_RAL4_95_MED2_trypsin_1.41211.41211.4         | DHSS | soluble hydrogenase small subunit  \n",
       "022016_RAL4_95_MED2_trypsin_2.50643.50643.4                         | groEL | chaperonin GroEL  \n",
       "022016_RAL4_95_MED2_trypsin_1.32793.32793.4  | fabF | 3-oxoacyl-[acyl-carrier-protein] synt...  \n",
       "022016_RAL4_95_MED2_trypsin_1.50751.50751.4                         | groEL | chaperonin GroEL  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# formerly, read in the replicates without precursor intensities and protein descriptions:\n",
    "\n",
    "# read the CSVs of each replicate into a datadrame we name 'comet' using the pandas read_csv function\n",
    "##comet1 = pd.read_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/RAL4_95_MED2_trypsin/TPP/RAL4_MED2_trypsin_1_PTMopt_Comet.csv\")\n",
    "##comet2 = pd.read_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/RAL4_95_MED2_trypsin/TPP/RAL4_MED2_trypsin_2_PTMopt_Comet.csv\")\n",
    "\n",
    "##frames = [comet1, comet2]\n",
    "\n",
    "# concatenate dataframes\n",
    "## cometdup = pd.concat(frames, sort=False)\n",
    "\n",
    "# now, reading in the combined csv that contains precursor intensities and protein descriptions\n",
    "cometdup = pd.read_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/RAL4_95_MED2_trypsin/TPP/ral_95_med2_trypsin_combine_quant.pep.sort.csv\", index_col='spectrum')\n",
    "\n",
    "# remove redundant rows\n",
    "comet = pd.DataFrame.drop_duplicates(cometdup)\n",
    "\n",
    "print(\"# redundant Comet peptides in combined dataframe\", len(cometdup))\n",
    "print(\"# nonredundant Comet peptides in combined dataframe\", len(comet))\n",
    "\n",
    "comet.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The peptide column has the residues before and after the tryptic terminii as well as masses of modified residues (e.g., 160.03 Da for carbamidomethylated cysteine). We want to make new columns with all that information and make a column with only the 'stripped' peptide sequence that's just amino acids - this we can then align against other sequences, for example. \n",
    "\n",
    "Modified residues were allowed for:\n",
    "\n",
    "- fixed carbamodimethylation of cysteine 57.021464 C\n",
    "- varialbe oxidation of methionine: 15.9949 M\n",
    "- variable deamidation of asparagine, glumatine: 0.984016 NQ\n",
    "- variable iron cation on lysine: 54.010565 K\n",
    "- variable methylation of lysine and arginine: 14.015650 KR\n",
    "\n",
    "We'll then write this manipulated dataframe to a new file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/millieginty/anaconda3/lib/python3.8/site-packages/pandas/core/ops/array_ops.py:253: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison\n",
      "  res_values = method(rvalues)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>xcorr</th>\n",
       "      <th>deltacn</th>\n",
       "      <th>expect</th>\n",
       "      <th>ions</th>\n",
       "      <th>peptide</th>\n",
       "      <th>protein</th>\n",
       "      <th>calc_neutral_pep_mass</th>\n",
       "      <th>precursor_intensity</th>\n",
       "      <th>protein_descr</th>\n",
       "      <th>L terminus</th>\n",
       "      <th>...</th>\n",
       "      <th>m-oxid</th>\n",
       "      <th>n-deam</th>\n",
       "      <th>q-deam</th>\n",
       "      <th>k-iron</th>\n",
       "      <th>k-meth</th>\n",
       "      <th>r-meth</th>\n",
       "      <th>stripped peptide</th>\n",
       "      <th>stripped length</th>\n",
       "      <th>ptm-total</th>\n",
       "      <th>NAAF num.</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>spectrum</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>022016_RAL4_95_MED2_trypsin_2.32518.32518.4</th>\n",
       "      <td>8.780</td>\n",
       "      <td>1.000</td>\n",
       "      <td>5.070000e-12</td>\n",
       "      <td>12/204</td>\n",
       "      <td>K.LAIDDSSINLDQVDYIN[115.03]AHGTSTTANDKNETSAIK.S</td>\n",
       "      <td>PMM1609</td>\n",
       "      <td>3734.7759</td>\n",
       "      <td>5250000.0</td>\n",
       "      <td>| fabF | 3-oxoacyl-[acyl-carrier-protein] synt...</td>\n",
       "      <td>K</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>LAIDDSSINLDQVDYINAHGTSTTANDKNETSAIK</td>\n",
       "      <td>35</td>\n",
       "      <td>1</td>\n",
       "      <td>150000.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>022016_RAL4_95_MED2_trypsin_1.41211.41211.4</th>\n",
       "      <td>8.768</td>\n",
       "      <td>1.000</td>\n",
       "      <td>1.420000e-09</td>\n",
       "      <td>22/156</td>\n",
       "      <td>K.LFADENHLSPAVTAIQIEDIDAEQFRK.N</td>\n",
       "      <td>PMM0035</td>\n",
       "      <td>3069.5407</td>\n",
       "      <td>2970000.0</td>\n",
       "      <td>| DHSS | soluble hydrogenase small subunit</td>\n",
       "      <td>K</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>LFADENHLSPAVTAIQIEDIDAEQFRK</td>\n",
       "      <td>27</td>\n",
       "      <td>0</td>\n",
       "      <td>110000.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>022016_RAL4_95_MED2_trypsin_2.50643.50643.4</th>\n",
       "      <td>8.599</td>\n",
       "      <td>0.583</td>\n",
       "      <td>2.140000e-12</td>\n",
       "      <td>22/156</td>\n",
       "      <td>R.SGLQNAASIAGM[147.04]VLTTEC[160.03]IVADLPEKK.D</td>\n",
       "      <td>PMM1436</td>\n",
       "      <td>2831.4409</td>\n",
       "      <td>7140000.0</td>\n",
       "      <td>| groEL | chaperonin GroEL</td>\n",
       "      <td>R</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>SGLQNAASIAGMIVADLPEKK</td>\n",
       "      <td>21</td>\n",
       "      <td>2</td>\n",
       "      <td>340000.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>022016_RAL4_95_MED2_trypsin_1.32793.32793.4</th>\n",
       "      <td>8.578</td>\n",
       "      <td>1.000</td>\n",
       "      <td>9.310000e-10</td>\n",
       "      <td>18/204</td>\n",
       "      <td>K.LAIDDSSIN[115.03]LDQVDYIN[115.03]AHGTSTTANDK...</td>\n",
       "      <td>PMM1609</td>\n",
       "      <td>3735.7599</td>\n",
       "      <td>6550000.0</td>\n",
       "      <td>| fabF | 3-oxoacyl-[acyl-carrier-protein] synt...</td>\n",
       "      <td>K</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>LAIDDSSINAHGTSTTANDKNETSAIK</td>\n",
       "      <td>27</td>\n",
       "      <td>2</td>\n",
       "      <td>242592.592593</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>022016_RAL4_95_MED2_trypsin_1.50751.50751.4</th>\n",
       "      <td>8.469</td>\n",
       "      <td>0.646</td>\n",
       "      <td>3.430000e-14</td>\n",
       "      <td>24/156</td>\n",
       "      <td>R.SGLQNAASIAGM[147.04]VLTTEC[160.03]IVADLPEKK.D</td>\n",
       "      <td>PMM1436</td>\n",
       "      <td>2831.4409</td>\n",
       "      <td>8290000.0</td>\n",
       "      <td>| groEL | chaperonin GroEL</td>\n",
       "      <td>R</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>SGLQNAASIAGMIVADLPEKK</td>\n",
       "      <td>21</td>\n",
       "      <td>2</td>\n",
       "      <td>394761.904762</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 42 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             xcorr  deltacn        expect  \\\n",
       "spectrum                                                                    \n",
       "022016_RAL4_95_MED2_trypsin_2.32518.32518.4  8.780    1.000  5.070000e-12   \n",
       "022016_RAL4_95_MED2_trypsin_1.41211.41211.4  8.768    1.000  1.420000e-09   \n",
       "022016_RAL4_95_MED2_trypsin_2.50643.50643.4  8.599    0.583  2.140000e-12   \n",
       "022016_RAL4_95_MED2_trypsin_1.32793.32793.4  8.578    1.000  9.310000e-10   \n",
       "022016_RAL4_95_MED2_trypsin_1.50751.50751.4  8.469    0.646  3.430000e-14   \n",
       "\n",
       "                                               ions  \\\n",
       "spectrum                                              \n",
       "022016_RAL4_95_MED2_trypsin_2.32518.32518.4  12/204   \n",
       "022016_RAL4_95_MED2_trypsin_1.41211.41211.4  22/156   \n",
       "022016_RAL4_95_MED2_trypsin_2.50643.50643.4  22/156   \n",
       "022016_RAL4_95_MED2_trypsin_1.32793.32793.4  18/204   \n",
       "022016_RAL4_95_MED2_trypsin_1.50751.50751.4  24/156   \n",
       "\n",
       "                                                                                       peptide  \\\n",
       "spectrum                                                                                         \n",
       "022016_RAL4_95_MED2_trypsin_2.32518.32518.4    K.LAIDDSSINLDQVDYIN[115.03]AHGTSTTANDKNETSAIK.S   \n",
       "022016_RAL4_95_MED2_trypsin_1.41211.41211.4                    K.LFADENHLSPAVTAIQIEDIDAEQFRK.N   \n",
       "022016_RAL4_95_MED2_trypsin_2.50643.50643.4    R.SGLQNAASIAGM[147.04]VLTTEC[160.03]IVADLPEKK.D   \n",
       "022016_RAL4_95_MED2_trypsin_1.32793.32793.4  K.LAIDDSSIN[115.03]LDQVDYIN[115.03]AHGTSTTANDK...   \n",
       "022016_RAL4_95_MED2_trypsin_1.50751.50751.4    R.SGLQNAASIAGM[147.04]VLTTEC[160.03]IVADLPEKK.D   \n",
       "\n",
       "                                             protein  calc_neutral_pep_mass  \\\n",
       "spectrum                                                                      \n",
       "022016_RAL4_95_MED2_trypsin_2.32518.32518.4  PMM1609              3734.7759   \n",
       "022016_RAL4_95_MED2_trypsin_1.41211.41211.4  PMM0035              3069.5407   \n",
       "022016_RAL4_95_MED2_trypsin_2.50643.50643.4  PMM1436              2831.4409   \n",
       "022016_RAL4_95_MED2_trypsin_1.32793.32793.4  PMM1609              3735.7599   \n",
       "022016_RAL4_95_MED2_trypsin_1.50751.50751.4  PMM1436              2831.4409   \n",
       "\n",
       "                                             precursor_intensity  \\\n",
       "spectrum                                                           \n",
       "022016_RAL4_95_MED2_trypsin_2.32518.32518.4            5250000.0   \n",
       "022016_RAL4_95_MED2_trypsin_1.41211.41211.4            2970000.0   \n",
       "022016_RAL4_95_MED2_trypsin_2.50643.50643.4            7140000.0   \n",
       "022016_RAL4_95_MED2_trypsin_1.32793.32793.4            6550000.0   \n",
       "022016_RAL4_95_MED2_trypsin_1.50751.50751.4            8290000.0   \n",
       "\n",
       "                                                                                 protein_descr  \\\n",
       "spectrum                                                                                         \n",
       "022016_RAL4_95_MED2_trypsin_2.32518.32518.4  | fabF | 3-oxoacyl-[acyl-carrier-protein] synt...   \n",
       "022016_RAL4_95_MED2_trypsin_1.41211.41211.4         | DHSS | soluble hydrogenase small subunit   \n",
       "022016_RAL4_95_MED2_trypsin_2.50643.50643.4                         | groEL | chaperonin GroEL   \n",
       "022016_RAL4_95_MED2_trypsin_1.32793.32793.4  | fabF | 3-oxoacyl-[acyl-carrier-protein] synt...   \n",
       "022016_RAL4_95_MED2_trypsin_1.50751.50751.4                         | groEL | chaperonin GroEL   \n",
       "\n",
       "                                            L terminus  ... m-oxid  n-deam  \\\n",
       "spectrum                                                ...                  \n",
       "022016_RAL4_95_MED2_trypsin_2.32518.32518.4          K  ...      0       1   \n",
       "022016_RAL4_95_MED2_trypsin_1.41211.41211.4          K  ...      0       0   \n",
       "022016_RAL4_95_MED2_trypsin_2.50643.50643.4          R  ...      1       0   \n",
       "022016_RAL4_95_MED2_trypsin_1.32793.32793.4          K  ...      0       2   \n",
       "022016_RAL4_95_MED2_trypsin_1.50751.50751.4          R  ...      1       0   \n",
       "\n",
       "                                             q-deam  k-iron  k-meth  r-meth  \\\n",
       "spectrum                                                                      \n",
       "022016_RAL4_95_MED2_trypsin_2.32518.32518.4       0       0       0       0   \n",
       "022016_RAL4_95_MED2_trypsin_1.41211.41211.4       0       0       0       0   \n",
       "022016_RAL4_95_MED2_trypsin_2.50643.50643.4       0       0       0       0   \n",
       "022016_RAL4_95_MED2_trypsin_1.32793.32793.4       0       0       0       0   \n",
       "022016_RAL4_95_MED2_trypsin_1.50751.50751.4       0       0       0       0   \n",
       "\n",
       "                                                                stripped peptide  \\\n",
       "spectrum                                                                           \n",
       "022016_RAL4_95_MED2_trypsin_2.32518.32518.4  LAIDDSSINLDQVDYINAHGTSTTANDKNETSAIK   \n",
       "022016_RAL4_95_MED2_trypsin_1.41211.41211.4          LFADENHLSPAVTAIQIEDIDAEQFRK   \n",
       "022016_RAL4_95_MED2_trypsin_2.50643.50643.4                SGLQNAASIAGMIVADLPEKK   \n",
       "022016_RAL4_95_MED2_trypsin_1.32793.32793.4          LAIDDSSINAHGTSTTANDKNETSAIK   \n",
       "022016_RAL4_95_MED2_trypsin_1.50751.50751.4                SGLQNAASIAGMIVADLPEKK   \n",
       "\n",
       "                                             stripped length  ptm-total  \\\n",
       "spectrum                                                                  \n",
       "022016_RAL4_95_MED2_trypsin_2.32518.32518.4               35          1   \n",
       "022016_RAL4_95_MED2_trypsin_1.41211.41211.4               27          0   \n",
       "022016_RAL4_95_MED2_trypsin_2.50643.50643.4               21          2   \n",
       "022016_RAL4_95_MED2_trypsin_1.32793.32793.4               27          2   \n",
       "022016_RAL4_95_MED2_trypsin_1.50751.50751.4               21          2   \n",
       "\n",
       "                                                 NAAF num.  \n",
       "spectrum                                                    \n",
       "022016_RAL4_95_MED2_trypsin_2.32518.32518.4  150000.000000  \n",
       "022016_RAL4_95_MED2_trypsin_1.41211.41211.4  110000.000000  \n",
       "022016_RAL4_95_MED2_trypsin_2.50643.50643.4  340000.000000  \n",
       "022016_RAL4_95_MED2_trypsin_1.32793.32793.4  242592.592593  \n",
       "022016_RAL4_95_MED2_trypsin_1.50751.50751.4  394761.904762  \n",
       "\n",
       "[5 rows x 42 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# get rid of rows where the xcorr is unavailable (usually 3 or so)\n",
    "comet = comet[comet.xcorr != '[unavailable]']\n",
    "\n",
    "# use str.strip with indexing by str[0] to add a column with the peptide's left terminus\n",
    "comet['L terminus'] = comet['peptide'].astype(str).str[0]\n",
    "\n",
    "# use str.strip with indexing by str[-1] to add a column with the peptide's left terminus\n",
    "comet['R terminus'] = comet['peptide'].str.strip().str[-1]\n",
    "\n",
    "# use a count function to enumerate the # of A's (alanines) in each peptide\n",
    "comet['A'] = comet['peptide'].str.count(\"A\")\n",
    "\n",
    "# use a count function to enumerate the # of C's (cysteines) in each peptide\n",
    "comet['C'] = comet['peptide'].str.count(\"C\")\n",
    "\n",
    "# use a count function to enumerate the # of D's (aspartic acids) in each peptide\n",
    "comet['D'] = comet['peptide'].str.count(\"D\")\n",
    "\n",
    "# use a count function to enumerate the # of E's (glutamic acids) in each peptide\n",
    "comet['E'] = comet['peptide'].str.count(\"E\")\n",
    "\n",
    "# use a count function to enumerate the # of F's (phenylalanines) in each peptide\n",
    "comet['F'] = comet['peptide'].str.count(\"F\")\n",
    "\n",
    "# use a count function to enumerate the # of G's (glycines) in each peptide\n",
    "comet['G'] = comet['peptide'].str.count(\"G\")\n",
    "\n",
    "# use a count function to enumerate the # of H's (histidines) in each peptide\n",
    "comet['H'] = comet['peptide'].str.count(\"H\")\n",
    "\n",
    "# use a count function to enumerate the # of I's (isoleucines) in each peptide\n",
    "# in comet output, there will be no isoleucines (they're lumped in with leucines)\n",
    "comet['I'] = comet['peptide'].str.count(\"I\")\n",
    "\n",
    "# use a count function to enumerate the # of K's (lysines) in each peptide\n",
    "comet['K'] = comet['peptide'].str.count(\"K\")\n",
    "\n",
    "# use a count function to enumerate the # of L's (leucines) in each peptide\n",
    "comet['L'] = comet['peptide'].str.count(\"L\")\n",
    "\n",
    "# use a count function to enumerate the # of M's (methionines) in each peptide\n",
    "comet['M'] = comet['peptide'].str.count(\"M\")\n",
    "\n",
    "# use a count function to enumerate the # of N's (asparagines) in each peptide\n",
    "comet['N'] = comet['peptide'].str.count(\"N\")\n",
    "\n",
    "# use a count function to enumerate the # of P's ([prolines]) in each peptide\n",
    "comet['P'] = comet['peptide'].str.count(\"P\")\n",
    "\n",
    "# use a count function to enumerate the # of Q's (glutamines) in each peptide\n",
    "comet['Q'] = comet['peptide'].str.count(\"Q\")\n",
    "\n",
    "# use a count function to enumerate the # of R's (arginines) in each peptide\n",
    "comet['R'] = comet['peptide'].str.count(\"R\")\n",
    "\n",
    "# use a count function to enumerate the # of S's (serines) in each peptide\n",
    "comet['S'] = comet['peptide'].str.count(\"S\")\n",
    "\n",
    "# use a count function to enumerate the # of T's (threonines) in each peptide\n",
    "comet['T'] = comet['peptide'].str.count(\"T\")\n",
    "\n",
    "# use a count function to enumerate the # of V's (valines) in each peptide\n",
    "comet['V'] = comet['peptide'].str.count(\"V\")\n",
    "\n",
    "# use a count function to enumerate the # of W's (tryptophans) in each peptide\n",
    "comet['W'] = comet['peptide'].str.count(\"W\")\n",
    "\n",
    "# use a count function to enumerate the # of Y's (tyrosines) in each peptide\n",
    "comet['Y'] = comet['peptide'].str.count(\"Y\")\n",
    "\n",
    "# use a count function to enumerate the # of ccarbamidomethylated C's in each peptide\n",
    "comet['c-carb'] = comet['peptide'].str.count(\"160.03\")\n",
    "\n",
    "# use a count function to enumerate the # of oxidized M's in each peptide\n",
    "comet['m-oxid'] = comet['peptide'].str.count(\"147.04\")\n",
    "\n",
    "# use a count function to enumerate the # of deamidated N's in each peptide\n",
    "comet['n-deam'] = comet['peptide'].str.count(\"115.03\")\n",
    "\n",
    "# use a count function to enumerate the # of deamidated Q's in each peptide\n",
    "comet['q-deam'] = comet['peptide'].str.count(\"129.04\")\n",
    "\n",
    "# use a count function to enumerate the # of iron adducted K's in each peptide\n",
    "comet['k-iron'] = comet['peptide'].str.count(\"182.11\")\n",
    "\n",
    "# use a count function to enumerate the # of methylated K's in each peptide\n",
    "comet['k-meth'] = comet['peptide'].str.count(\"142.11\")\n",
    "\n",
    "# use a count function to enumerate the # of methylated R's in each peptide\n",
    "comet['r-meth'] = comet['peptide'].str.count(\"170.12\")\n",
    "\n",
    "# create a column with 'stripped' peptide sequences using strip\n",
    "comet['stripped peptide'] = comet['peptide'].str[2:].str[:-2].str.replace(r\"\\[.*\\]\",\"\")\n",
    "\n",
    "# add a column with the stripped peptide length (number of AAs)\n",
    "comet['stripped length'] = comet['stripped peptide'].apply(len)\n",
    "\n",
    "# total the number of modifications in sequence\n",
    "comet['ptm-total'] = comet['c-carb'] + comet['m-oxid'] + comet['n-deam'] + comet['q-deam'] + comet['k-iron'] + comet['k-meth'] + comet['r-meth']\n",
    "\n",
    "# calculate the NAAF numerator for later NAAF normalization\n",
    "comet['NAAF num.'] = comet['precursor_intensity'] / comet['stripped length']\n",
    "\n",
    "# write modified dataframe to new txt file, plus 'unfiltered'\n",
    "comet.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/RAL4_95_MED2_trypsin/TPP/RAL4_MED2_trypsin_1_PTMopt_Comet_unfiltered.csv\")\n",
    "\n",
    "# check out the results\n",
    "comet.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Calculating the false discovery rate (% FDR)\n",
    "\n",
    "### Filtering PSMs > a selected XCorr value and exporting peptides"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# real Comet PSMs 54797\n",
      "# decoy Comet PSMs 10738\n",
      "False discovery rate =  19.59596328266146\n"
     ]
    }
   ],
   "source": [
    "# Let's separate out the decoy hits from the good ones\n",
    "\n",
    "cometpmm = comet[~comet['protein'].str.contains(\"DECOY\")]\n",
    "cometdec = comet[comet['protein'].str.contains(\"DECOY\")]\n",
    "\n",
    "# how many PSM that are only PMM (proteins in the database)?\n",
    "\n",
    "print(\"# real Comet PSMs\", len(cometpmm))\n",
    "\n",
    "# compared to how many PSMs containing decoys?\n",
    "\n",
    "print(\"# decoy Comet PSMs\", len(cometdec))\n",
    "\n",
    "# calculate the bulk FDR (all PSMs so let's not beat ourselves up)\n",
    "\n",
    "r = len(cometpmm)\n",
    "d = len(cometdec)\n",
    "\n",
    "FDR = d/r*100\n",
    "\n",
    "print(\"False discovery rate = \", FDR)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# real Comet PSMs 35931\n",
      "# decoy Comet PSMs 1270\n",
      "False discovery rate =  3.4138867234751755\n"
     ]
    }
   ],
   "source": [
    "# keep only peptides  >2.5 XCorr\n",
    "# need to convert Xcorr column from strings to numeric so we can use loc\n",
    "comet['xcorr'] = pd.to_numeric(comet['xcorr'])\n",
    "\n",
    "comet25 = comet.loc[comet['xcorr'] >= 2.5]\n",
    "\n",
    "# What's the FDR?\n",
    "\n",
    "# Let's separate out the decoy hits from the good ones\n",
    "\n",
    "cometpmm25 = comet25[~comet25['protein'].str.contains(\"DECOY\")]\n",
    "cometdec25 = comet25[comet25['protein'].str.contains(\"DECOY\")]\n",
    "\n",
    "# how many PSM that are only PMM (proteins in the database)?\n",
    "\n",
    "print(\"# real Comet PSMs\", len(cometpmm25))\n",
    "\n",
    "# compared to how many PSMs containing decoys?\n",
    "\n",
    "print(\"# decoy Comet PSMs\", len(cometdec25))\n",
    "\n",
    "# calculate the FDR \n",
    "\n",
    "r = len(cometpmm25)\n",
    "d = len(cometdec25)\n",
    "\n",
    "FDR = d/(d+r)*100\n",
    "\n",
    "print(\"False discovery rate = \", FDR)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# real Comet PSMs 26923\n",
      "# decoy Comet PSMs 258\n",
      "False discovery rate =  0.9491924506088812\n"
     ]
    }
   ],
   "source": [
    "# keep only peptides  >3 XCorr\n",
    "# need to convert Xcorr column from strings to numeric so we can use loc\n",
    "comet['xcorr'] = pd.to_numeric(comet['xcorr'])\n",
    "\n",
    "comet3 = comet.loc[comet['xcorr'] >= 3]\n",
    "\n",
    "# What's the FDR?\n",
    "\n",
    "# Let's separate out the decoy hits from the good ones\n",
    "\n",
    "cometpmm3 = comet3[~comet3['protein'].str.contains(\"DECOY\")]\n",
    "cometdec3 = comet3[comet3['protein'].str.contains(\"DECOY\")]\n",
    "\n",
    "# how many PSM that are only PMM (proteins in the database)?\n",
    "\n",
    "print(\"# real Comet PSMs\", len(cometpmm3))\n",
    "\n",
    "# compared to how many PSMs containing decoys?\n",
    "\n",
    "print(\"# decoy Comet PSMs\", len(cometdec3))\n",
    "\n",
    "# calculate the FDR \n",
    "\n",
    "r = len(cometpmm3)\n",
    "d = len(cometdec3)\n",
    "\n",
    "FDR = d/(d+r)*100\n",
    "\n",
    "print(\"False discovery rate = \", FDR)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Exporting peptides from a XCorr > 2.5 and XCorr > 3 thresholds:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# redundant Comet peptides >2.5 XCorr 35931\n",
      "# nonredundant Comet peptides >2.5 XCOrr 12283\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>stripped peptide</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>spectrum</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>022016_RAL4_95_MED2_trypsin_2.32518.32518.4</th>\n",
       "      <td>LAIDDSSINLDQVDYINAHGTSTTANDKNETSAIK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>022016_RAL4_95_MED2_trypsin_1.41211.41211.4</th>\n",
       "      <td>LFADENHLSPAVTAIQIEDIDAEQFRK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>022016_RAL4_95_MED2_trypsin_2.50643.50643.4</th>\n",
       "      <td>SGLQNAASIAGMIVADLPEKK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>022016_RAL4_95_MED2_trypsin_1.32793.32793.4</th>\n",
       "      <td>LAIDDSSINAHGTSTTANDKNETSAIK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>022016_RAL4_95_MED2_trypsin_1.50751.50751.4</th>\n",
       "      <td>SGLQNAASIAGMIVADLPEKK</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                stripped peptide\n",
       "spectrum                                                                        \n",
       "022016_RAL4_95_MED2_trypsin_2.32518.32518.4  LAIDDSSINLDQVDYINAHGTSTTANDKNETSAIK\n",
       "022016_RAL4_95_MED2_trypsin_1.41211.41211.4          LFADENHLSPAVTAIQIEDIDAEQFRK\n",
       "022016_RAL4_95_MED2_trypsin_2.50643.50643.4                SGLQNAASIAGMIVADLPEKK\n",
       "022016_RAL4_95_MED2_trypsin_1.32793.32793.4          LAIDDSSINAHGTSTTANDKNETSAIK\n",
       "022016_RAL4_95_MED2_trypsin_1.50751.50751.4                SGLQNAASIAGMIVADLPEKK"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# keep only peptides  >2.5 XCorr\n",
    "# need to convert Xcorr column from strings to numeric so we can use loc\n",
    "comet['xcorr'] = pd.to_numeric(comet['xcorr'])\n",
    "\n",
    "comet25 = comet.loc[comet['xcorr'] >= 2.5]\n",
    "\n",
    "# Let's separate out the decoy hits from the good ones\n",
    "\n",
    "cometpmm25 = comet25[~comet25['protein'].str.contains(\"DECOY\")]\n",
    "cometdec25 = comet25[comet25['protein'].str.contains(\"DECOY\")]\n",
    "\n",
    "\n",
    "# keep only peptide column \n",
    "pep25 = cometpmm25[[\"stripped peptide\"]]\n",
    "\n",
    "# write altered dataframe to new txt file\n",
    "# used header and index parameters to get rid of 'Peptide' header and the indexing\n",
    "\n",
    "pep25.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/RAL4_95_MED2_trypsin/TPP/RAL4_MED2_trypsin_combine_PTMopt_Comet_stripped_peptides_2.5XCorr.txt\", header=False, index=False)\n",
    "\n",
    "# removing redundancy\n",
    "pep25dedup = pd.DataFrame.drop_duplicates(pep25)\n",
    "\n",
    "print(\"# redundant Comet peptides >2.5 XCorr\", len(pep25))\n",
    "print(\"# nonredundant Comet peptides >2.5 XCOrr\", len(pep25dedup))\n",
    "\n",
    "pep25.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# redundant Comet peptides >3 XCorr 26923\n",
      "# nonredundant Comet peptides >3 XCOrr 9213\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>stripped peptide</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>spectrum</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>022016_RAL4_95_MED2_trypsin_2.32518.32518.4</th>\n",
       "      <td>LAIDDSSINLDQVDYINAHGTSTTANDKNETSAIK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>022016_RAL4_95_MED2_trypsin_1.41211.41211.4</th>\n",
       "      <td>LFADENHLSPAVTAIQIEDIDAEQFRK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>022016_RAL4_95_MED2_trypsin_2.50643.50643.4</th>\n",
       "      <td>SGLQNAASIAGMIVADLPEKK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>022016_RAL4_95_MED2_trypsin_1.32793.32793.4</th>\n",
       "      <td>LAIDDSSINAHGTSTTANDKNETSAIK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>022016_RAL4_95_MED2_trypsin_1.50751.50751.4</th>\n",
       "      <td>SGLQNAASIAGMIVADLPEKK</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                stripped peptide\n",
       "spectrum                                                                        \n",
       "022016_RAL4_95_MED2_trypsin_2.32518.32518.4  LAIDDSSINLDQVDYINAHGTSTTANDKNETSAIK\n",
       "022016_RAL4_95_MED2_trypsin_1.41211.41211.4          LFADENHLSPAVTAIQIEDIDAEQFRK\n",
       "022016_RAL4_95_MED2_trypsin_2.50643.50643.4                SGLQNAASIAGMIVADLPEKK\n",
       "022016_RAL4_95_MED2_trypsin_1.32793.32793.4          LAIDDSSINAHGTSTTANDKNETSAIK\n",
       "022016_RAL4_95_MED2_trypsin_1.50751.50751.4                SGLQNAASIAGMIVADLPEKK"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# keep only peptides  >3 XCorr\n",
    "# need to convert Xcorr column from strings to numeric so we can use loc\n",
    "comet['xcorr'] = pd.to_numeric(comet['xcorr'])\n",
    "\n",
    "comet3 = comet.loc[comet['xcorr'] >= 3]\n",
    "\n",
    "# Let's separate out the decoy hits from the good ones\n",
    "\n",
    "cometpmm3 = comet3[~comet3['protein'].str.contains(\"DECOY\")]\n",
    "cometdec3 = comet3[comet3['protein'].str.contains(\"DECOY\")]\n",
    "\n",
    "# export the whole table for Comet XCorr > 3\n",
    "cometpmm3.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/RAL4_95_MED2_trypsin/TPP/RAL4_MED2_trypsin_combine_PTMopt_Comet_3XCorr_noDECOY.csv\")\n",
    "\n",
    "# keep only peptide column \n",
    "pep3 = cometpmm3[[\"stripped peptide\"]]\n",
    "\n",
    "# write altered dataframe to new txt file\n",
    "# used header and index parameters to get rid of 'Peptide' header and the indexing\n",
    "\n",
    "pep3.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/RAL4_95_MED2_trypsin/TPP/RAL4_MED2_trypsin_combine_PTMopt_Comet_stripped_peptides_3XCorr.txt\", header=False, index=False)\n",
    "\n",
    "# removing redundancy\n",
    "pep3dedup = pd.DataFrame.drop_duplicates(pep3)\n",
    "\n",
    "print(\"# redundant Comet peptides >3 XCorr\", len(pep3))\n",
    "print(\"# nonredundant Comet peptides >3 XCOrr\", len(pep3dedup))\n",
    "\n",
    "pep3.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## NAAF correction and exporting files with AA and PTM totals:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>A</th>\n",
       "      <th>C</th>\n",
       "      <th>D</th>\n",
       "      <th>E</th>\n",
       "      <th>F</th>\n",
       "      <th>G</th>\n",
       "      <th>H</th>\n",
       "      <th>I</th>\n",
       "      <th>K</th>\n",
       "      <th>L</th>\n",
       "      <th>...</th>\n",
       "      <th>Total area</th>\n",
       "      <th>Total length</th>\n",
       "      <th>% C w/ carb.</th>\n",
       "      <th>% M w/ oxid</th>\n",
       "      <th>% N w/ deam</th>\n",
       "      <th>% Q w/ deam</th>\n",
       "      <th>% K w/ iron</th>\n",
       "      <th>% K w/ meth</th>\n",
       "      <th>% R w/ meth</th>\n",
       "      <th>NAAF denom.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>sample total</th>\n",
       "      <td>34390</td>\n",
       "      <td>3829</td>\n",
       "      <td>29556</td>\n",
       "      <td>36962</td>\n",
       "      <td>16265</td>\n",
       "      <td>34557</td>\n",
       "      <td>6414</td>\n",
       "      <td>35470</td>\n",
       "      <td>41629</td>\n",
       "      <td>49572</td>\n",
       "      <td>...</td>\n",
       "      <td>1.236939e+11</td>\n",
       "      <td>412111</td>\n",
       "      <td>0.92818</td>\n",
       "      <td>0.292821</td>\n",
       "      <td>0.066992</td>\n",
       "      <td>0.067926</td>\n",
       "      <td>0.007423</td>\n",
       "      <td>0.003459</td>\n",
       "      <td>0.003377</td>\n",
       "      <td>300146.975419</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows Ã— 37 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  A     C      D      E      F      G     H      I      K  \\\n",
       "sample total  34390  3829  29556  36962  16265  34557  6414  35470  41629   \n",
       "\n",
       "                  L  ...    Total area  Total length  % C w/ carb.  \\\n",
       "sample total  49572  ...  1.236939e+11        412111       0.92818   \n",
       "\n",
       "              % M w/ oxid  % N w/ deam  % Q w/ deam  % K w/ iron  % K w/ meth  \\\n",
       "sample total     0.292821     0.066992     0.067926     0.007423     0.003459   \n",
       "\n",
       "              % R w/ meth    NAAF denom.  \n",
       "sample total     0.003377  300146.975419  \n",
       "\n",
       "[1 rows x 37 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# made a new dataframe that contains the sums of certain columns in the stripped peptide dataframe above\n",
    "# choosing the XCorr > 3 filtered results (no decoys)\n",
    "\n",
    "index = ['sample total']\n",
    "\n",
    "data = {'A': cometpmm3['A'].sum(),\n",
    "        'C': cometpmm3['C'].sum(),\n",
    "        'D': cometpmm3['D'].sum(),\n",
    "        'E': cometpmm3['E'].sum(),\n",
    "        'F': cometpmm3['F'].sum(),\n",
    "        'G': cometpmm3['G'].sum(),\n",
    "        'H': cometpmm3['H'].sum(),\n",
    "        'I': cometpmm3['I'].sum(),\n",
    "        'K': cometpmm3['K'].sum(),\n",
    "        'L': cometpmm3['L'].sum(),\n",
    "        'M': cometpmm3['M'].sum(),\n",
    "        'N': cometpmm3['N'].sum(),\n",
    "        'P': cometpmm3['P'].sum(),\n",
    "        'Q': cometpmm3['Q'].sum(),\n",
    "        'R': cometpmm3['R'].sum(),\n",
    "        'S': cometpmm3['S'].sum(),\n",
    "        'T': cometpmm3['T'].sum(),\n",
    "        'V': cometpmm3['V'].sum(),\n",
    "        'W': cometpmm3['W'].sum(),\n",
    "        'Y': cometpmm3['Y'].sum(),\n",
    "        'c-carb': cometpmm3['c-carb'].sum(),\n",
    "        'm-oxid': cometpmm3['m-oxid'].sum(),\n",
    "        'n-deam': cometpmm3['n-deam'].sum(),\n",
    "        'q-deam': cometpmm3['q-deam'].sum(),\n",
    "        'k-iron': cometpmm3['k-iron'].sum(),\n",
    "        'k-meth': cometpmm3['k-meth'].sum(),\n",
    "        'r-meth': cometpmm3['r-meth'].sum(),\n",
    "        'Total area': cometpmm3['precursor_intensity'].sum(),\n",
    "        'Total length': cometpmm3['stripped length'].sum()\n",
    "       }\n",
    "\n",
    "totalcometpmm3 = pd.DataFrame(data, columns=['A', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'K', \\\n",
    "                                           'L', 'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'V', \\\n",
    "                                           'W', 'Y', 'c-carb', 'm-oxid', 'n-deam', \\\n",
    "                                           'q-deam', 'k-iron', 'k-meth', 'r-meth', \\\n",
    "                                          'Total area', 'Total length'], index=index)\n",
    "\n",
    "# calculate percentage of C's with carb (should be 1.0)\n",
    "totalcometpmm3['% C w/ carb.'] = totalcometpmm3['c-carb'] / totalcometpmm3['C'] \n",
    "\n",
    "# calculate percentage of M's that are oxidized\n",
    "totalcometpmm3['% M w/ oxid'] = totalcometpmm3['m-oxid'] / totalcometpmm3['M'] \n",
    "\n",
    "# calculate percentage of N's that are deamidated\n",
    "totalcometpmm3['% N w/ deam'] = totalcometpmm3['n-deam'] / totalcometpmm3['N'] \n",
    "\n",
    "# calculate percentage of Q's that are deamidated\n",
    "totalcometpmm3['% Q w/ deam'] = totalcometpmm3['q-deam'] / totalcometpmm3['Q'] \n",
    "\n",
    "# calculate percentage of K's that are hydroxylated\n",
    "totalcometpmm3['% K w/ iron'] = totalcometpmm3['k-iron'] / totalcometpmm3['K'] \n",
    "\n",
    "# calculate percentage of K's that are methylated\n",
    "totalcometpmm3['% K w/ meth'] = totalcometpmm3['k-meth'] / totalcometpmm3['K'] \n",
    "\n",
    "# calculate percentage of R's that are methylated\n",
    "totalcometpmm3['% R w/ meth'] = totalcometpmm3['r-meth'] / totalcometpmm3['R'] \n",
    "\n",
    "# calculate NAAF denominator for all peptides in dataset i\n",
    "totalcometpmm3['NAAF denom.'] = totalcometpmm3['Total area'] / totalcometpmm3['Total length']\n",
    "\n",
    "# write modified dataframe to new txt file, same name + totals\n",
    "totalcometpmm3.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/RAL4_95_MED2_trypsin/TPP/RAL95_MED2_trypsin_combine_cometpmm3_totals.csv\")\n",
    "\n",
    "totalcometpmm3.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-15-963b09fbd510>:7: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  cometpmm3['NAAF factor'] = (cometpmm3['NAAF num.'])/NAAF3\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>stripped peptide</th>\n",
       "      <th>NAAF factor</th>\n",
       "      <th>A</th>\n",
       "      <th>C</th>\n",
       "      <th>D</th>\n",
       "      <th>E</th>\n",
       "      <th>F</th>\n",
       "      <th>G</th>\n",
       "      <th>H</th>\n",
       "      <th>K</th>\n",
       "      <th>...</th>\n",
       "      <th>V-NAAF3</th>\n",
       "      <th>W-NAAF3</th>\n",
       "      <th>Y-NAAF3</th>\n",
       "      <th>ccarb-NAAF3</th>\n",
       "      <th>moxid-NAAF3</th>\n",
       "      <th>ndeam-NAAF3</th>\n",
       "      <th>qdeam-NAAF3</th>\n",
       "      <th>kiron-NAAF3</th>\n",
       "      <th>kmeth-NAAF3</th>\n",
       "      <th>rmeth-NAAF3</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>spectrum</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>022016_RAL4_95_MED2_trypsin_2.32518.32518.4</th>\n",
       "      <td>LAIDDSSINLDQVDYINAHGTSTTANDKNETSAIK</td>\n",
       "      <td>0.499755</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>...</td>\n",
       "      <td>0.499755</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.499755</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.499755</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>022016_RAL4_95_MED2_trypsin_1.41211.41211.4</th>\n",
       "      <td>LFADENHLSPAVTAIQIEDIDAEQFRK</td>\n",
       "      <td>0.366487</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>0.366487</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>022016_RAL4_95_MED2_trypsin_2.50643.50643.4</th>\n",
       "      <td>SGLQNAASIAGMIVADLPEKK</td>\n",
       "      <td>1.132778</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2.265557</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.132778</td>\n",
       "      <td>1.132778</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>022016_RAL4_95_MED2_trypsin_1.32793.32793.4</th>\n",
       "      <td>LAIDDSSINAHGTSTTANDKNETSAIK</td>\n",
       "      <td>0.808246</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>...</td>\n",
       "      <td>0.808246</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.808246</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.616492</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>022016_RAL4_95_MED2_trypsin_1.50751.50751.4</th>\n",
       "      <td>SGLQNAASIAGMIVADLPEKK</td>\n",
       "      <td>1.315229</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>2.630457</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.315229</td>\n",
       "      <td>1.315229</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 56 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                stripped peptide  \\\n",
       "spectrum                                                                           \n",
       "022016_RAL4_95_MED2_trypsin_2.32518.32518.4  LAIDDSSINLDQVDYINAHGTSTTANDKNETSAIK   \n",
       "022016_RAL4_95_MED2_trypsin_1.41211.41211.4          LFADENHLSPAVTAIQIEDIDAEQFRK   \n",
       "022016_RAL4_95_MED2_trypsin_2.50643.50643.4                SGLQNAASIAGMIVADLPEKK   \n",
       "022016_RAL4_95_MED2_trypsin_1.32793.32793.4          LAIDDSSINAHGTSTTANDKNETSAIK   \n",
       "022016_RAL4_95_MED2_trypsin_1.50751.50751.4                SGLQNAASIAGMIVADLPEKK   \n",
       "\n",
       "                                             NAAF factor  A  C  D  E  F  G  H  \\\n",
       "spectrum                                                                        \n",
       "022016_RAL4_95_MED2_trypsin_2.32518.32518.4     0.499755  4  0  5  1  0  1  1   \n",
       "022016_RAL4_95_MED2_trypsin_1.41211.41211.4     0.366487  4  0  3  3  2  0  1   \n",
       "022016_RAL4_95_MED2_trypsin_2.50643.50643.4     1.132778  4  1  2  2  0  2  0   \n",
       "022016_RAL4_95_MED2_trypsin_1.32793.32793.4     0.808246  4  0  5  1  0  1  1   \n",
       "022016_RAL4_95_MED2_trypsin_1.50751.50751.4     1.315229  4  1  2  2  0  2  0   \n",
       "\n",
       "                                             K  ...   V-NAAF3  W-NAAF3  \\\n",
       "spectrum                                        ...                      \n",
       "022016_RAL4_95_MED2_trypsin_2.32518.32518.4  3  ...  0.499755      0.0   \n",
       "022016_RAL4_95_MED2_trypsin_1.41211.41211.4  2  ...  0.366487      0.0   \n",
       "022016_RAL4_95_MED2_trypsin_2.50643.50643.4  2  ...  2.265557      0.0   \n",
       "022016_RAL4_95_MED2_trypsin_1.32793.32793.4  3  ...  0.808246      0.0   \n",
       "022016_RAL4_95_MED2_trypsin_1.50751.50751.4  2  ...  2.630457      0.0   \n",
       "\n",
       "                                              Y-NAAF3  ccarb-NAAF3  \\\n",
       "spectrum                                                             \n",
       "022016_RAL4_95_MED2_trypsin_2.32518.32518.4  0.499755     0.000000   \n",
       "022016_RAL4_95_MED2_trypsin_1.41211.41211.4  0.000000     0.000000   \n",
       "022016_RAL4_95_MED2_trypsin_2.50643.50643.4  0.000000     1.132778   \n",
       "022016_RAL4_95_MED2_trypsin_1.32793.32793.4  0.808246     0.000000   \n",
       "022016_RAL4_95_MED2_trypsin_1.50751.50751.4  0.000000     1.315229   \n",
       "\n",
       "                                             moxid-NAAF3  ndeam-NAAF3  \\\n",
       "spectrum                                                                \n",
       "022016_RAL4_95_MED2_trypsin_2.32518.32518.4     0.000000     0.499755   \n",
       "022016_RAL4_95_MED2_trypsin_1.41211.41211.4     0.000000     0.000000   \n",
       "022016_RAL4_95_MED2_trypsin_2.50643.50643.4     1.132778     0.000000   \n",
       "022016_RAL4_95_MED2_trypsin_1.32793.32793.4     0.000000     1.616492   \n",
       "022016_RAL4_95_MED2_trypsin_1.50751.50751.4     1.315229     0.000000   \n",
       "\n",
       "                                             qdeam-NAAF3  kiron-NAAF3  \\\n",
       "spectrum                                                                \n",
       "022016_RAL4_95_MED2_trypsin_2.32518.32518.4          0.0          0.0   \n",
       "022016_RAL4_95_MED2_trypsin_1.41211.41211.4          0.0          0.0   \n",
       "022016_RAL4_95_MED2_trypsin_2.50643.50643.4          0.0          0.0   \n",
       "022016_RAL4_95_MED2_trypsin_1.32793.32793.4          0.0          0.0   \n",
       "022016_RAL4_95_MED2_trypsin_1.50751.50751.4          0.0          0.0   \n",
       "\n",
       "                                             kmeth-NAAF3  rmeth-NAAF3  \n",
       "spectrum                                                               \n",
       "022016_RAL4_95_MED2_trypsin_2.32518.32518.4          0.0          0.0  \n",
       "022016_RAL4_95_MED2_trypsin_1.41211.41211.4          0.0          0.0  \n",
       "022016_RAL4_95_MED2_trypsin_2.50643.50643.4          0.0          0.0  \n",
       "022016_RAL4_95_MED2_trypsin_1.32793.32793.4          0.0          0.0  \n",
       "022016_RAL4_95_MED2_trypsin_1.50751.50751.4          0.0          0.0  \n",
       "\n",
       "[5 rows x 56 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# use the calculated NAAF factor (from above) to caluclate the NAAF \n",
    "# NAAF: normalized normalized area abundance factor\n",
    "\n",
    "NAAF3 = 300146.975419\n",
    "\n",
    "# use NAAF >XCorr 3 to get NAAF\n",
    "cometpmm3['NAAF factor'] = (cometpmm3['NAAF num.'])/NAAF3\n",
    "\n",
    "# make a dataframe that contains only what we need: sequences, AAs, PTMs\n",
    "cometpmm3_AA = cometpmm3[['stripped peptide', 'NAAF factor', 'A', 'C', 'D', 'E', 'F', 'G', 'H', 'K', 'I', 'L', \\\n",
    "                                'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'V', 'W', 'Y', 'c-carb', 'm-oxid', 'n-deam', \\\n",
    "                                'q-deam', 'k-iron', 'k-meth', 'r-meth']].copy()\n",
    "\n",
    "# multiply the NAAF3 factor by the AAs to normalize its abundance by peak area and peptide length\n",
    "\n",
    "cometpmm3_AA['A-NAAF3'] = cometpmm3_AA['A'] * cometpmm3['NAAF factor']\n",
    "cometpmm3_AA['C-NAAF3'] = cometpmm3_AA['C'] * cometpmm3['NAAF factor']\n",
    "cometpmm3_AA['D-NAAF3'] = cometpmm3_AA['D'] * cometpmm3['NAAF factor']\n",
    "cometpmm3_AA['E-NAAF3'] = cometpmm3_AA['E'] * cometpmm3['NAAF factor']\n",
    "cometpmm3_AA['F-NAAF3'] = cometpmm3_AA['F'] * cometpmm3['NAAF factor']\n",
    "cometpmm3_AA['G-NAAF3'] = cometpmm3_AA['G'] * cometpmm3['NAAF factor']\n",
    "cometpmm3_AA['H-NAAF3'] = cometpmm3_AA['H'] * cometpmm3['NAAF factor']\n",
    "cometpmm3_AA['K-NAAF3'] = cometpmm3_AA['K'] * cometpmm3['NAAF factor']\n",
    "cometpmm3_AA['I-NAAF3'] = cometpmm3_AA['I'] * cometpmm3['NAAF factor']\n",
    "cometpmm3_AA['L-NAAF3'] = cometpmm3_AA['L'] * cometpmm3['NAAF factor']\n",
    "cometpmm3_AA['M-NAAF3'] = cometpmm3_AA['M'] * cometpmm3['NAAF factor']\n",
    "cometpmm3_AA['N-NAAF3'] = cometpmm3_AA['N'] * cometpmm3['NAAF factor']\n",
    "cometpmm3_AA['P-NAAF3'] = cometpmm3_AA['P'] * cometpmm3['NAAF factor']\n",
    "cometpmm3_AA['Q-NAAF3'] = cometpmm3_AA['Q'] * cometpmm3['NAAF factor']\n",
    "cometpmm3_AA['R-NAAF3'] = cometpmm3_AA['R'] * cometpmm3['NAAF factor']\n",
    "cometpmm3_AA['S-NAAF3'] = cometpmm3_AA['S'] * cometpmm3['NAAF factor']\n",
    "cometpmm3_AA['T-NAAF3'] = cometpmm3_AA['T'] * cometpmm3['NAAF factor']\n",
    "cometpmm3_AA['V-NAAF3'] = cometpmm3_AA['V'] * cometpmm3['NAAF factor']\n",
    "cometpmm3_AA['W-NAAF3'] = cometpmm3_AA['W'] * cometpmm3['NAAF factor']\n",
    "cometpmm3_AA['Y-NAAF3'] = cometpmm3_AA['Y'] * cometpmm3['NAAF factor']\n",
    "\n",
    "# multiply the NAAF3 factor by the PTMs normalize its abundance by peak area and peptide length\n",
    "\n",
    "cometpmm3_AA['ccarb-NAAF3'] = cometpmm3_AA['c-carb'] * cometpmm3_AA['NAAF factor']\n",
    "cometpmm3_AA['moxid-NAAF3'] = cometpmm3_AA['m-oxid'] * cometpmm3_AA['NAAF factor']\n",
    "cometpmm3_AA['ndeam-NAAF3'] = cometpmm3_AA['n-deam'] * cometpmm3_AA['NAAF factor']\n",
    "cometpmm3_AA['qdeam-NAAF3'] = cometpmm3_AA['q-deam'] * cometpmm3_AA['NAAF factor']\n",
    "cometpmm3_AA['kiron-NAAF3'] = cometpmm3_AA['k-iron'] * cometpmm3_AA['NAAF factor']\n",
    "cometpmm3_AA['kmeth-NAAF3'] = cometpmm3_AA['k-meth'] * cometpmm3_AA['NAAF factor']\n",
    "cometpmm3_AA['rmeth-NAAF3'] = cometpmm3_AA['r-meth'] * cometpmm3_AA['NAAF factor']\n",
    "\n",
    "# write the dataframe to a new csv\n",
    "cometpmm3_AA.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/RAL4_95_MED2_trypsin/TPP/RAL4_MED2_combine_Comet3_AA_NAAF.csv\")\n",
    "\n",
    "cometpmm3_AA.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### All XCorr - Visualizing the results:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# making evenly spaced bins for the Xcorr data based on the min and max, called above\n",
    "bins = [0, 0.5, 1, 1.5, 2, 2.5, 3, 3.5, 4, 4.5, 5, 5.5, 6, 6.5, 7, 7.5, 8, 8.5, 9]\n",
    "labels = ['0-0.5', '0.5-1', '1-1.5', '1.5-2', '2-2.5', '2.5-3', '3-3.5', '3.5-4', '4-4.5', '4.5-5', '5-5.5', '5.5-6', '6-6.5', '6.5-7', '7-7.5', '7.5-8', '8-8.5', '8.5-9']\n",
    "\n",
    "# use pandas cut function to do the binning itself\n",
    "comet['binned'] = pd.cut(comet['xcorr'], bins=bins, labels=labels)\n",
    "\n",
    "# bar plots of binned PTM data\n",
    "\n",
    "index = ['0-0.5', '0.5-1', '1-1.5', '1.5-2', '2-2.5', '2.5-3', '3-3.5', '3.5-4', '4-4.5', '4.5-5', '5-5.5', '5.5-6', '6-6.5', '6.5-7', '7-7.5', '7.5-8', '8-8.5', '8.5-9']\n",
    "data = {'Total PTMs': [comet.groupby('binned')['ptm-total'].sum()['0-0.5'], comet.groupby('binned')['ptm-total'].sum()['0.5-1'], comet.groupby('binned')['ptm-total'].sum()['1-1.5'], comet.groupby('binned')['ptm-total'].sum()['1.5-2'], comet.groupby('binned')['ptm-total'].sum()['2-2.5'], comet.groupby('binned')['ptm-total'].sum()['2.5-3'], comet.groupby('binned')['ptm-total'].sum()['3-3.5'], comet.groupby('binned')['ptm-total'].sum()['3.5-4'], comet.groupby('binned')['ptm-total'].sum()['4-4.5'], comet.groupby('binned')['ptm-total'].sum()['4.5-5'], comet.groupby('binned')['ptm-total'].sum()['5-5.5'], comet.groupby('binned')['ptm-total'].sum()['5.5-6'], comet.groupby('binned')['ptm-total'].sum()['6-6.5'], comet.groupby('binned')['ptm-total'].sum()['6.5-7'], comet.groupby('binned')['ptm-total'].sum()['7-7.5'], comet.groupby('binned')['ptm-total'].sum()['7.5-8'], comet.groupby('binned')['ptm-total'].sum()['8-8.5'], comet.groupby('binned')['ptm-total'].sum()['8.5-9']],\n",
    "        'Cys carb.': [comet.groupby('binned')['c-carb'].sum()['0-0.5'], comet.groupby('binned')['c-carb'].sum()['0.5-1'], comet.groupby('binned')['c-carb'].sum()['1-1.5'], comet.groupby('binned')['c-carb'].sum()['1.5-2'], comet.groupby('binned')['c-carb'].sum()['2-2.5'], comet.groupby('binned')['c-carb'].sum()['2.5-3'], comet.groupby('binned')['c-carb'].sum()['3-3.5'], comet.groupby('binned')['c-carb'].sum()['3.5-4'], comet.groupby('binned')['c-carb'].sum()['4-4.5'], comet.groupby('binned')['c-carb'].sum()['4.5-5'], comet.groupby('binned')['c-carb'].sum()['5-5.5'], comet.groupby('binned')['c-carb'].sum()['5.5-6'], comet.groupby('binned')['c-carb'].sum()['6-6.5'], comet.groupby('binned')['c-carb'].sum()['6.5-7'], comet.groupby('binned')['c-carb'].sum()['7-7.5'], comet.groupby('binned')['c-carb'].sum()['7.5-8'], comet.groupby('binned')['c-carb'].sum()['8-8.5'], comet.groupby('binned')['c-carb'].sum()['8.5-9']],\n",
    "        'Met oxi.': [comet.groupby('binned')['m-oxid'].sum()['0-0.5'], comet.groupby('binned')['m-oxid'].sum()['0.5-1'], comet.groupby('binned')['m-oxid'].sum()['1-1.5'], comet.groupby('binned')['m-oxid'].sum()['1.5-2'], comet.groupby('binned')['m-oxid'].sum()['2-2.5'], comet.groupby('binned')['m-oxid'].sum()['2.5-3'], comet.groupby('binned')['m-oxid'].sum()['3-3.5'], comet.groupby('binned')['m-oxid'].sum()['3.5-4'], comet.groupby('binned')['m-oxid'].sum()['4-4.5'], comet.groupby('binned')['m-oxid'].sum()['4.5-5'], comet.groupby('binned')['m-oxid'].sum()['5-5.5'], comet.groupby('binned')['m-oxid'].sum()['5.5-6'], comet.groupby('binned')['m-oxid'].sum()['6-6.5'], comet.groupby('binned')['m-oxid'].sum()['6.5-7'], comet.groupby('binned')['m-oxid'].sum()['7-7.5'], comet.groupby('binned')['m-oxid'].sum()['7.5-8'], comet.groupby('binned')['m-oxid'].sum()['8-8.5'], comet.groupby('binned')['m-oxid'].sum()['8.5-9']],\n",
    "        'Asp deam.': [comet.groupby('binned')['n-deam'].sum()['0-0.5'], comet.groupby('binned')['n-deam'].sum()['0.5-1'], comet.groupby('binned')['n-deam'].sum()['1-1.5'], comet.groupby('binned')['n-deam'].sum()['1.5-2'], comet.groupby('binned')['n-deam'].sum()['2-2.5'], comet.groupby('binned')['n-deam'].sum()['2.5-3'], comet.groupby('binned')['n-deam'].sum()['3-3.5'], comet.groupby('binned')['n-deam'].sum()['3.5-4'], comet.groupby('binned')['n-deam'].sum()['4-4.5'], comet.groupby('binned')['n-deam'].sum()['4.5-5'], comet.groupby('binned')['n-deam'].sum()['5-5.5'], comet.groupby('binned')['n-deam'].sum()['5.5-6'], comet.groupby('binned')['n-deam'].sum()['6-6.5'], comet.groupby('binned')['n-deam'].sum()['6.5-7'], comet.groupby('binned')['n-deam'].sum()['7-7.5'], comet.groupby('binned')['n-deam'].sum()['7.5-8'], comet.groupby('binned')['n-deam'].sum()['8-8.5'], comet.groupby('binned')['n-deam'].sum()['8.5-9']],\n",
    "        'Glut deam.': [comet.groupby('binned')['q-deam'].sum()['0-0.5'], comet.groupby('binned')['q-deam'].sum()['0.5-1'], comet.groupby('binned')['q-deam'].sum()['1-1.5'], comet.groupby('binned')['q-deam'].sum()['1.5-2'], comet.groupby('binned')['q-deam'].sum()['2-2.5'], comet.groupby('binned')['q-deam'].sum()['2.5-3'], comet.groupby('binned')['q-deam'].sum()['3-3.5'], comet.groupby('binned')['q-deam'].sum()['3.5-4'], comet.groupby('binned')['q-deam'].sum()['4-4.5'], comet.groupby('binned')['q-deam'].sum()['4.5-5'], comet.groupby('binned')['q-deam'].sum()['5-5.5'], comet.groupby('binned')['q-deam'].sum()['5.5-6'], comet.groupby('binned')['q-deam'].sum()['6-6.5'], comet.groupby('binned')['q-deam'].sum()['6.5-7'], comet.groupby('binned')['q-deam'].sum()['7-7.5'], comet.groupby('binned')['q-deam'].sum()['7.5-8'], comet.groupby('binned')['q-deam'].sum()['8-8.5'], comet.groupby('binned')['q-deam'].sum()['8.5-9']],\n",
    "        'Lys iron': [comet.groupby('binned')['k-iron'].sum()['0-0.5'], comet.groupby('binned')['k-iron'].sum()['0.5-1'], comet.groupby('binned')['k-iron'].sum()['1-1.5'], comet.groupby('binned')['k-iron'].sum()['1.5-2'], comet.groupby('binned')['k-iron'].sum()['2-2.5'], comet.groupby('binned')['k-iron'].sum()['2.5-3'], comet.groupby('binned')['k-iron'].sum()['3-3.5'], comet.groupby('binned')['k-iron'].sum()['3.5-4'], comet.groupby('binned')['k-iron'].sum()['4-4.5'], comet.groupby('binned')['k-iron'].sum()['4.5-5'], comet.groupby('binned')['k-iron'].sum()['5-5.5'], comet.groupby('binned')['k-iron'].sum()['5.5-6'], comet.groupby('binned')['k-iron'].sum()['6-6.5'], comet.groupby('binned')['k-iron'].sum()['6.5-7'], comet.groupby('binned')['k-iron'].sum()['7-7.5'], comet.groupby('binned')['k-iron'].sum()['7.5-8'], comet.groupby('binned')['k-iron'].sum()['8-8.5'], comet.groupby('binned')['k-iron'].sum()['8.5-9']],\n",
    "        'Lys meth.': [comet.groupby('binned')['k-meth'].sum()['0-0.5'], comet.groupby('binned')['k-meth'].sum()['0.5-1'], comet.groupby('binned')['k-meth'].sum()['1-1.5'], comet.groupby('binned')['k-meth'].sum()['1.5-2'], comet.groupby('binned')['k-meth'].sum()['2-2.5'], comet.groupby('binned')['k-meth'].sum()['2.5-3'], comet.groupby('binned')['k-meth'].sum()['3-3.5'], comet.groupby('binned')['k-meth'].sum()['3.5-4'], comet.groupby('binned')['k-meth'].sum()['4-4.5'], comet.groupby('binned')['k-meth'].sum()['4.5-5'], comet.groupby('binned')['k-meth'].sum()['5-5.5'], comet.groupby('binned')['k-meth'].sum()['5.5-6'], comet.groupby('binned')['k-meth'].sum()['6-6.5'], comet.groupby('binned')['k-meth'].sum()['6.5-7'], comet.groupby('binned')['k-meth'].sum()['7-7.5'], comet.groupby('binned')['k-meth'].sum()['7.5-8'], comet.groupby('binned')['k-meth'].sum()['8-8.5'], comet.groupby('binned')['k-meth'].sum()['8.5-9']],\n",
    "        'Arg meth.': [comet.groupby('binned')['r-meth'].sum()['0-0.5'], comet.groupby('binned')['r-meth'].sum()['0.5-1'], comet.groupby('binned')['r-meth'].sum()['1-1.5'], comet.groupby('binned')['r-meth'].sum()['1.5-2'], comet.groupby('binned')['r-meth'].sum()['2-2.5'], comet.groupby('binned')['r-meth'].sum()['2.5-3'], comet.groupby('binned')['r-meth'].sum()['3-3.5'], comet.groupby('binned')['r-meth'].sum()['3.5-4'], comet.groupby('binned')['r-meth'].sum()['4-4.5'], comet.groupby('binned')['r-meth'].sum()['4.5-5'], comet.groupby('binned')['r-meth'].sum()['5-5.5'], comet.groupby('binned')['r-meth'].sum()['5.5-6'], comet.groupby('binned')['r-meth'].sum()['6-6.5'], comet.groupby('binned')['r-meth'].sum()['6.5-7'], comet.groupby('binned')['r-meth'].sum()['7-7.5'], comet.groupby('binned')['r-meth'].sum()['7.5-8'], comet.groupby('binned')['r-meth'].sum()['8-8.5'], comet.groupby('binned')['r-meth'].sum()['8.5-9']]\n",
    "        }\n",
    "\n",
    "cometbin = pd.DataFrame(data, columns=['Total PTMs','Cys carb.','Met oxi.','Asp deam.', 'Glut deam.', 'Lys iron', 'Lys meth.', 'Arg meth.'], index=index)\n",
    "\n",
    "ax1 = cometbin.plot.bar(y='Total PTMs', rot=45)\n",
    "ax1.set_title('Total PTMs')\n",
    "\n",
    "ax2 = cometbin.plot.bar(y='Cys carb.', rot=45)\n",
    "ax2.set_title('Cysteine carbamidomethylation')\n",
    "\n",
    "ax3 = cometbin.plot.bar(y='Met oxi.', rot=45)\n",
    "ax3.set_title('Methionine oxidation')\n",
    "\n",
    "ax4 = cometbin.plot.bar(y='Asp deam.', rot=45)\n",
    "ax4.set_title('Asparagine deamidation')\n",
    "\n",
    "ax5 = cometbin.plot.bar(y='Glut deam.', rot=45)\n",
    "ax5.set_title('Glutamine deamidation')\n",
    "\n",
    "ax6 = cometbin.plot.bar(y='Lys iron', rot=45)\n",
    "ax6.set_title('Lysine iron adduct')\n",
    "\n",
    "ax7 = cometbin.plot.bar(y='Lys meth.', rot=45)\n",
    "ax7.set_title('Lysine methylation')\n",
    "\n",
    "ax7 = cometbin.plot.bar(y='Arg meth.', rot=45)\n",
    "ax7.set_title('Arginine methylation')\n",
    "\n",
    "#plt.savefig('/home/millieginty/Documents/git-repos/2017-etnp/analyses/pronovo-2020/pronovo-2020-ptm/MED4_trypsin1_Comet_PTMopt.png')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# histogram of stripped peptide lengths\n",
    "\n",
    "#plt.rcdefaults()\n",
    "#fig, ax = plt.subplots()\n",
    "\n",
    "comet.plot(y='stripped length', kind = 'hist', bins = 20, title = 'Peptide length')\n",
    "plt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# histogram of total peptide amounts\n",
    "\n",
    "#plt.rcdefaults()\n",
    "#fig, ax = plt.subplots()\n",
    "\n",
    "comet.plot(y='ptm-total', kind = 'hist', bins = 20, title = 'PTMs/sequence')\n",
    "plt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from matplotlib import pyplot\n",
    "# density plot of xcorr vs length (idea from https://python-graph-gallery.com/85-density-plot-with-matplotlib/)\n",
    "\n",
    "# read in data\n",
    "x = comet['xcorr']\n",
    "y = comet['stripped length']\n",
    " \n",
    "# evaluate a gaussian kernel density estimation (KDE) on a regular grid of nbins x nbins over data extents\n",
    "nbins=300\n",
    "k = kde.gaussian_kde([x,y])\n",
    "xi, yi = np.mgrid[x.min():x.max():nbins*1j, y.min():y.max():nbins*1j]\n",
    "zi = k(np.vstack([xi.flatten(), yi.flatten()]))\n",
    " \n",
    "# make the plot\n",
    "plt.pcolormesh(xi, yi, zi.reshape(xi.shape))\n",
    "plt.colorbar()\n",
    "plt.show() "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# plot with density of xcorr vs length and xcorr vs total ptms\n",
    "plt.figure()\n",
    "\n",
    "# read in data\n",
    "x = comet['xcorr']\n",
    "y = comet['stripped length']\n",
    "\n",
    "a = comet['xcorr']\n",
    "b = comet['ptm-total']\n",
    "\n",
    "# evaluate a gaussian kernel density estimation (KDE) on a regular grid of nbins x nbins over data extents\n",
    "nbins=300\n",
    "k = kde.gaussian_kde([x,y])\n",
    "xi, yi = np.mgrid[x.min():x.max():nbins*1j, y.min():y.max():nbins*1j]\n",
    "zi = k(np.vstack([xi.flatten(), yi.flatten()]))\n",
    "\n",
    "c = kde.gaussian_kde([a,b])\n",
    "ai, bi = np.mgrid[a.min():a.max():nbins*1j, b.min():b.max():nbins*1j]\n",
    "di = k(np.vstack([ai.flatten(), bi.flatten()]))\n",
    "\n",
    "# density plot of length vs xcorr\n",
    "plt.subplot(221)\n",
    "plt.pcolormesh(xi, yi, zi.reshape(xi.shape))\n",
    "plt.title('logit')\n",
    "plt.colorbar()\n",
    "\n",
    "# density plot of xcorr vs ptms\n",
    "plt.subplot(222)\n",
    "plt.pcolormesh(ai, bi, di.reshape(ai.shape))\n",
    "plt.title('Combined MED4 Comet XCorr vs total PTMs')\n",
    "plt.colorbar()\n",
    "\n",
    "\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### XCorr > 2.5 - Visualizing the results:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "# making evenly spaced bins for the Xcorr data based on the min and max, called above\n",
    "bins = [2.5, 3, 3.5, 4, 4.5, 5, 5.5, 6, 6.5, 7, 7.5, 8, 8.5, 9]\n",
    "labels = ['2.5-3', '3-3.5', '3.5-4', '4-4.5', '4.5-5', '5-5.5', '5.5-6', '6-6.5', '6.5-7', '7-7.5', '7.5-8', '8-8.5', '8.5-9']\n",
    "\n",
    "# use pandas cut function to do the binning itself\n",
    "comet25['binned'] = pd.cut(comet25['xcorr'], bins=bins, labels=labels)\n",
    "\n",
    "# bar plots of binned PTM data\n",
    "\n",
    "index = ['2.5-3', '3-3.5', '3.5-4', '4-4.5', '4.5-5', '5-5.5', '5.5-6', '6-6.5', '6.5-7', '7-7.5', '7.5-8', '8-8.5', '8.5-9']\n",
    "data = {'Total PTMs': [comet25.groupby('binned')['ptm-total'].sum()['2.5-3'], comet25.groupby('binned')['ptm-total'].sum()['3-3.5'], comet25.groupby('binned')['ptm-total'].sum()['3.5-4'], comet25.groupby('binned')['ptm-total'].sum()['4-4.5'], comet25.groupby('binned')['ptm-total'].sum()['4.5-5'], comet25.groupby('binned')['ptm-total'].sum()['5-5.5'], comet25.groupby('binned')['ptm-total'].sum()['5.5-6'], comet25.groupby('binned')['ptm-total'].sum()['6-6.5'], comet25.groupby('binned')['ptm-total'].sum()['6.5-7'], comet25.groupby('binned')['ptm-total'].sum()['7-7.5'], comet25.groupby('binned')['ptm-total'].sum()['7.5-8'], comet25.groupby('binned')['ptm-total'].sum()['8-8.5'], comet25.groupby('binned')['ptm-total'].sum()['8.5-9']],\n",
    "        'Cys carb.': [comet25.groupby('binned')['c-carb'].sum()['2.5-3'], comet25.groupby('binned')['c-carb'].sum()['3-3.5'], comet25.groupby('binned')['c-carb'].sum()['3.5-4'], comet25.groupby('binned')['c-carb'].sum()['4-4.5'], comet25.groupby('binned')['c-carb'].sum()['4.5-5'], comet25.groupby('binned')['c-carb'].sum()['5-5.5'], comet25.groupby('binned')['c-carb'].sum()['5.5-6'], comet25.groupby('binned')['c-carb'].sum()['6-6.5'], comet25.groupby('binned')['c-carb'].sum()['6.5-7'], comet25.groupby('binned')['c-carb'].sum()['7-7.5'], comet25.groupby('binned')['c-carb'].sum()['7.5-8'], comet25.groupby('binned')['c-carb'].sum()['8-8.5'], comet25.groupby('binned')['c-carb'].sum()['8.5-9']],\n",
    "        'Met oxi.': [comet25.groupby('binned')['m-oxid'].sum()['2.5-3'], comet25.groupby('binned')['m-oxid'].sum()['3-3.5'], comet25.groupby('binned')['m-oxid'].sum()['3.5-4'], comet25.groupby('binned')['m-oxid'].sum()['4-4.5'], comet25.groupby('binned')['m-oxid'].sum()['4.5-5'], comet25.groupby('binned')['m-oxid'].sum()['5-5.5'], comet25.groupby('binned')['m-oxid'].sum()['5.5-6'], comet25.groupby('binned')['m-oxid'].sum()['6-6.5'], comet25.groupby('binned')['m-oxid'].sum()['6.5-7'], comet25.groupby('binned')['m-oxid'].sum()['7-7.5'], comet25.groupby('binned')['m-oxid'].sum()['7.5-8'], comet25.groupby('binned')['m-oxid'].sum()['8-8.5'], comet25.groupby('binned')['m-oxid'].sum()['8.5-9']],\n",
    "        'Asp deam.': [comet25.groupby('binned')['n-deam'].sum()['2.5-3'], comet25.groupby('binned')['n-deam'].sum()['3-3.5'], comet25.groupby('binned')['n-deam'].sum()['3.5-4'], comet25.groupby('binned')['n-deam'].sum()['4-4.5'], comet25.groupby('binned')['n-deam'].sum()['4.5-5'], comet25.groupby('binned')['n-deam'].sum()['5-5.5'], comet25.groupby('binned')['n-deam'].sum()['5.5-6'], comet25.groupby('binned')['n-deam'].sum()['6-6.5'], comet25.groupby('binned')['n-deam'].sum()['6.5-7'], comet25.groupby('binned')['n-deam'].sum()['7-7.5'], comet25.groupby('binned')['n-deam'].sum()['7.5-8'], comet25.groupby('binned')['n-deam'].sum()['8-8.5'], comet25.groupby('binned')['n-deam'].sum()['8.5-9']],\n",
    "        'Glut deam.': [comet25.groupby('binned')['q-deam'].sum()['2.5-3'], comet25.groupby('binned')['q-deam'].sum()['3-3.5'], comet25.groupby('binned')['q-deam'].sum()['3.5-4'], comet25.groupby('binned')['q-deam'].sum()['4-4.5'], comet25.groupby('binned')['q-deam'].sum()['4.5-5'], comet25.groupby('binned')['q-deam'].sum()['5-5.5'], comet25.groupby('binned')['q-deam'].sum()['5.5-6'], comet25.groupby('binned')['q-deam'].sum()['6-6.5'], comet25.groupby('binned')['q-deam'].sum()['6.5-7'], comet25.groupby('binned')['q-deam'].sum()['7-7.5'], comet25.groupby('binned')['q-deam'].sum()['7.5-8'], comet25.groupby('binned')['q-deam'].sum()['8-8.5'], comet25.groupby('binned')['q-deam'].sum()['8.5-9']],\n",
    "        'Lys iron': [comet25.groupby('binned')['k-iron'].sum()['2.5-3'], comet25.groupby('binned')['k-iron'].sum()['3-3.5'], comet25.groupby('binned')['k-iron'].sum()['3.5-4'], comet25.groupby('binned')['k-iron'].sum()['4-4.5'], comet25.groupby('binned')['k-iron'].sum()['4.5-5'], comet25.groupby('binned')['k-iron'].sum()['5-5.5'], comet25.groupby('binned')['k-iron'].sum()['5.5-6'], comet25.groupby('binned')['k-iron'].sum()['6-6.5'], comet25.groupby('binned')['k-iron'].sum()['6.5-7'], comet25.groupby('binned')['k-iron'].sum()['7-7.5'], comet25.groupby('binned')['k-iron'].sum()['7.5-8'], comet25.groupby('binned')['k-iron'].sum()['8-8.5'], comet25.groupby('binned')['k-iron'].sum()['8.5-9']],\n",
    "        'Lys meth.': [comet25.groupby('binned')['k-meth'].sum()['2.5-3'], comet25.groupby('binned')['k-meth'].sum()['3-3.5'], comet25.groupby('binned')['k-meth'].sum()['3.5-4'], comet25.groupby('binned')['k-meth'].sum()['4-4.5'], comet25.groupby('binned')['k-meth'].sum()['4.5-5'], comet25.groupby('binned')['k-meth'].sum()['5-5.5'], comet25.groupby('binned')['k-meth'].sum()['5.5-6'], comet25.groupby('binned')['k-meth'].sum()['6-6.5'], comet25.groupby('binned')['k-meth'].sum()['6.5-7'], comet25.groupby('binned')['k-meth'].sum()['7-7.5'], comet25.groupby('binned')['k-meth'].sum()['7.5-8'], comet25.groupby('binned')['k-meth'].sum()['8-8.5'], comet25.groupby('binned')['k-meth'].sum()['8.5-9']],\n",
    "        'Arg meth.': [comet25.groupby('binned')['r-meth'].sum()['2.5-3'], comet25.groupby('binned')['r-meth'].sum()['3-3.5'], comet25.groupby('binned')['r-meth'].sum()['3.5-4'], comet25.groupby('binned')['r-meth'].sum()['4-4.5'], comet25.groupby('binned')['r-meth'].sum()['4.5-5'], comet25.groupby('binned')['r-meth'].sum()['5-5.5'], comet25.groupby('binned')['r-meth'].sum()['5.5-6'], comet25.groupby('binned')['r-meth'].sum()['6-6.5'], comet25.groupby('binned')['r-meth'].sum()['6.5-7'], comet25.groupby('binned')['r-meth'].sum()['7-7.5'], comet25.groupby('binned')['r-meth'].sum()['7.5-8'], comet25.groupby('binned')['r-meth'].sum()['8-8.5'], comet25.groupby('binned')['r-meth'].sum()['8.5-9']]\n",
    "        }\n",
    "\n",
    "comet25bin = pd.DataFrame(data, columns=['Total PTMs','Cys carb.','Met oxi.','Asp deam.', 'Glut deam.', 'Lys iron', 'Lys meth.', 'Arg meth.'], index=index)\n",
    "\n",
    "ax1 = comet25bin.plot.bar(y='Total PTMs', rot=45)\n",
    "ax1.set_title('Total PTMs')\n",
    "\n",
    "ax2 = comet25bin.plot.bar(y='Cys carb.', rot=45)\n",
    "ax2.set_title('Cysteine carbamidomethylation')\n",
    "\n",
    "ax3 = comet25bin.plot.bar(y='Met oxi.', rot=45)\n",
    "ax3.set_title('Methionine oxidation')\n",
    "\n",
    "ax4 = comet25bin.plot.bar(y='Asp deam.', rot=45)\n",
    "ax4.set_title('Asparagine deamidation')\n",
    "\n",
    "ax5 = comet25bin.plot.bar(y='Glut deam.', rot=45)\n",
    "ax5.set_title('Glutamine deamidation')\n",
    "\n",
    "ax6 = comet25bin.plot.bar(y='Lys iron', rot=45)\n",
    "ax6.set_title('Lysine iron adduct')\n",
    "\n",
    "ax7 = comet25bin.plot.bar(y='Lys meth.', rot=45)\n",
    "ax7.set_title('Lysine methylation')\n",
    "\n",
    "ax7 = comet25bin.plot.bar(y='Arg meth.', rot=45)\n",
    "ax7.set_title('Arginine methylation')\n",
    "\n",
    "#plt.savefig('/home/millieginty/Documents/git-repos/2017-etnp/analyses/pronovo-2020/pronovo-2020-ptm/MED4_trypsin1_comet25_PTMopt.png')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# histogram of stripped peptide lengths\n",
    "\n",
    "#plt.rcdefaults()\n",
    "#fig, ax = plt.subplots()\n",
    "\n",
    "comet25.plot(y='stripped length', kind = 'hist', bins = 20, title = 'Peptide length')\n",
    "plt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# histogram of total peptide amounts\n",
    "\n",
    "#plt.rcdefaults()\n",
    "#fig, ax = plt.subplots()\n",
    "\n",
    "comet25.plot(y='ptm-total', kind = 'hist', bins = 20, title = 'PTMs/sequence')\n",
    "plt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from matplotlib import pyplot\n",
    "# density plot of xcorr vs length (idea from https://python-graph-gallery.com/85-density-plot-with-matplotlib/)\n",
    "\n",
    "# read in data\n",
    "x = comet25['xcorr']\n",
    "y = comet25['stripped length']\n",
    " \n",
    "# evaluate a gaussian kernel density estimation (KDE) on a regular grid of nbins x nbins over data extents\n",
    "nbins=300\n",
    "k = kde.gaussian_kde([x,y])\n",
    "xi, yi = np.mgrid[x.min():x.max():nbins*1j, y.min():y.max():nbins*1j]\n",
    "zi = k(np.vstack([xi.flatten(), yi.flatten()]))\n",
    " \n",
    "# make the plot\n",
    "plt.pcolormesh(xi, yi, zi.reshape(xi.shape))\n",
    "plt.colorbar()\n",
    "plt.show() "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# now we have the stripped peptide csvs and txt files in the same data dir:\n",
    "!ls /home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/RAL4_95_MED2_trypsin/TPP/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
